pkgs/nitori_message/lib/src/parser.dart

```diff
@@ -212,14 +200,39 @@ class Parser {
         match = attributeRegex.firstMatch(attributesStr);
       }
 
+      if (!isClosingTag && !isSelfClosingTag) {
+        // Parse attributes and children
+        while (tokenizer.hasNext() && !tokenizer.peek()!.startsWith('</')) {
+          var child = parse(parentTagName: tagName);
+          if (child != null) {
+            children.add(child);
+          }
+        }
+        // Skip the closing tag
+        tokenizer.next();
+      }
+
       return _createTag(tagName, attributes, children);
     } else {
+      var t = parentTagName != null && _shouldNotTrim(parentTagName)
+          // trim the leading and trailing newlines
+          ? token
+              .replaceFirst(RegExp(r'^\n'), '')
+              .replaceFirst(RegExp(r'\n$'), '')
```

Comment:
```suggestion
          ? token
              .replaceFirst(RegExp(r'\n$'), '')
```

So an ideal changes like this? @dezhishen